packages/flutter/lib/src/widgets/overscroll_indicator.dart

```diff
@@ -637,18 +644,285 @@ class _GlowingOverscrollIndicatorPainter extends CustomPainter {
+  /// [ScrollBehavior.buildOverscrollIndicator] method when opted-in using the
+  /// [ScrollBehavior.updatedAndroidOverscrollIndicator] flag. In this case
+  /// the child is usually the one provided as an argument to that method.
+  final Widget? child;
+
+  @override
+  State<StretchingOverscrollIndicator> createState() => _StretchingOverscrollIndicatorState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(EnumProperty<AxisDirection>('axisDirection', axisDirection));
+  }
+}
+
+class _StretchingOverscrollIndicatorState extends State<StretchingOverscrollIndicator> with TickerProviderStateMixin {
+  _StretchController? _stretchController;
+  ScrollNotification? _lastNotification;
+  OverscrollNotification? _lastOverscrollNotification;
+  bool _accepted = true;
+
+  @override
+  void initState() {
+    super.initState();
+    _stretchController = _StretchController(vsync: this);
+  }
+
+  bool _handleScrollNotification(ScrollNotification notification) {
+    if (!widget.notificationPredicate(notification))
+      return false;
+
+    if (notification is OverscrollNotification) {
+      _lastOverscrollNotification = notification;
+      if (_lastNotification.runtimeType != OverscrollNotification) {
+        final OverscrollIndicatorNotification confirmationNotification = OverscrollIndicatorNotification(leading: notification.overscroll < 0.0);
+        confirmationNotification.dispatch(context);
+        _accepted = confirmationNotification._accepted;
+      }
+
+      assert(notification.metrics.axis == widget.axis);
+      if (_accepted) {
+        if (notification.velocity != 0.0) {
+          assert(notification.dragDetails == null);
+          _stretchController!.absorbImpact(notification.velocity.abs());
+        } else {
+          assert(notification.overscroll != 0.0);
+          if (notification.dragDetails != null) {
+            _stretchController!.pull(notification.overscroll.abs() / notification.metrics.viewportDimension);
+          }
+        }
+      }
+    } else if (notification is ScrollEndNotification && notification.dragDetails != null
+        || notification is ScrollUpdateNotification && notification.dragDetails != null) {
+      _stretchController!.scrollEnd();
+    }
+    _lastNotification = notification;
+    return false;
+  }
+
+  @override
+  void dispose() {
+    _stretchController!.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return NotificationListener<ScrollNotification>(
+      onNotification: _handleScrollNotification,
+      child: AnimatedBuilder(
+        animation: _stretchController!,
+        builder: (BuildContext context, Widget? child) {
+          final double stretch = _stretchController?.value ?? 0;
+          double x = 1.0;
+          double y = 1.0;
+          late AlignmentDirectional alignment;
```

Comment:
I believe you could just declare this as final instead of late. Dart should recognize that this gets assigned in all code passes.